src/pyramid/predicates.py

```diff
@@ -276,6 +276,19 @@ def __call__(self, context, request):
         return False
 
 
+class IsAuthenticatedPredicate:
+    def __init__(self, val, config):
+        self.val = val
+
+    def text(self):
+        return "is_authenticated = %r" % (self.val,)
+
+    phash = text
+
+    def __call__(self, context, request):
+        return bool(request.is_authenticated) is self.val
```

Comment:
Is `request.is_authenticated` already a boolean? If so, then is wrapping with `bool(...)` necessary? Or should `self.val` be wrapped, i.e., `bool(self.val)`?

Reply:
The code was `request.is_authenticated == self.val` before, but I changed it to match XhrPredicate.  I don’t understand the reason (maybe because `request.is_xhr` is not controlled by Pyramid but webob?), just thought the style should match.  I can easily revert.

Reply:
either way is fine to me but if I were writing it myself I would've used `==`